test/WebJobs.Script.Tests.Integration/ProxyEndToEndTests.cs

```diff
@@ -3,52 +3,105 @@
 
 using System;
 using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
 using System.Linq;
 using System.Net;
 using System.Net.Http;
 using System.Threading.Tasks;
-using Microsoft.Azure.AppService.Proxy.Client;
-using Microsoft.Azure.WebJobs.Script.Description;
-using Microsoft.Extensions.Logging;
-using Moq;
+using System.Web.Http;
+using Microsoft.Azure.WebJobs.Script.Config;
+using Microsoft.Azure.WebJobs.Script.WebHost;
 using Xunit;
 
 namespace Microsoft.Azure.WebJobs.Script.Tests
 {
-    public class ProxyEndToEndTests : EndToEndTestsBase<ProxyEndToEndTests.TestFixture>
+    public class ProxyEndToEndTests : IClassFixture<ProxyEndToEndTests.TestFixture>
     {
-        public ProxyEndToEndTests(TestFixture fixture) : base(fixture)
+        private readonly ScriptSettingsManager _settingsManager;
+        private TestFixture _fixture;
+
+        public ProxyEndToEndTests(TestFixture fixture)
         {
+            _fixture = fixture;
+            _settingsManager = ScriptSettingsManager.Instance;
         }
 
         [Fact]
         public async Task Proxy_Invoke_Succeeds()
         {
-            HttpRequestMessage request = new HttpRequestMessage
-            {
-                RequestUri = new Uri(string.Format("http://localhost/mymockhttp")),
-                Method = HttpMethod.Get,
-            };
+            HttpResponseMessage response = await _fixture.HttpClient.GetAsync($"/mymockhttp");
 
-            request.SetConfiguration(Fixture.RequestConfiguration);
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+            Assert.Equal(response.Headers.GetValues("myversion").ToArray()[0], "123");
+        }
 
-            Dictionary<string, object> arguments = new Dictionary<string, object>
-            {
-                { "req", request },
-            };
-            await Fixture.Host.CallAsync("localFunction", arguments);
+        [Fact]
+        public async Task Local_Function_Call()
+        {
+            HttpResponseMessage response = await _fixture.HttpClient.GetAsync($"/myproxy");
+
+            string content = await response.Content.ReadAsStringAsync();
 
-            HttpResponseMessage response = (HttpResponseMessage)request.Properties[ScriptConstants.AzureFunctionsHttpResponseKey];
             Assert.Equal(HttpStatusCode.OK, response.StatusCode);
-            Assert.Equal(response.Headers.GetValues("myversion").ToArray()[0], "123");
+            Assert.True(content.StartsWith("Time:"));
+        }
+
+        [Theory]
+        [InlineData("test.txt")]
```

Comment:
Consider adding more extensions like html, js, css

Reply:
Would probably make sense to add a `.svc` extension, just for the sake of covering the CRI condition.

Reply:
nvm, I see it now